analysis_driver/pipelines/projects.py

```diff
@@ -28,35 +32,39 @@ def build_pipeline(dataset):
     genotype_gvcfs = GenotypeGVCFs(dataset=dataset, gVCFs=gvcf_files, reference=reference)
     relatedness = Relatedness(dataset=dataset, previous_stages=[genotype_gvcfs])
     peddy = Peddy(dataset=dataset, ids=sample_ids, previous_stages=[genotype_gvcfs])
-    parse = ParseRelatedness(dataset=dataset, parse_method='parse_both', previous_stages=[relatedness, peddy])
-    output = Output(dataset=dataset, previous_stages=[relatedness, peddy, parse])
+    parse = ParseRelatedness(dataset=dataset, ids=sample_ids, parse_method='parse_both', previous_stages=[relatedness, peddy])
+    md5 = MD5Sum(dataset=dataset, previous_stages=[parse])
+    output = Output(dataset=dataset, previous_stages=[md5])
     cleanup = Cleanup(dataset=dataset, previous_stages=[output])
     return cleanup
 
 
+class MD5Sum(segmentation.Stage):
+    def _run(self):
+        dir_with_linked_files = os.path.join(self.job_dir, 'relatedness_outfiles')
+        return executor.execute(
+            *[bash_commands.md5sum(os.path.join(dir_with_linked_files, f)) for f in os.listdir(dir_with_linked_files)],
+            job_name='md5sum',
+            working_dir=self.job_dir,
+            cpus=1,
+            mem=2,
+            log_commands=False
+        ).join()
+
+
 class Output(segmentation.Stage):
     def _run(self):
+        dir_with_linked_files = os.path.join(self.job_dir, 'relatedness_outfiles')
+        os.makedirs(dir_with_linked_files, exist_ok=True)
+        output_cfg = OutputFileConfiguration('project_process')
 
-        dir_with_output_files = os.path.join(self.job_dir, 'relatedness_outfiles')
-        os.makedirs(dir_with_output_files, exist_ok=True)
-        output_files_cfg = OutputFileConfiguration('project_process')
-
-        for symlink_file in output_files_cfg.content.values():
-            source = os.path.join(
-                self.job_dir,
-                os.path.join(*symlink_file['location']),
-                symlink_file['basename'].format(project_id=self.dataset.name)
-            )
-
-            symlink_path = os.path.join(dir_with_output_files, symlink_file['basename'].format(project_id=self.dataset.name))
-            if os.path.isfile(source):
-                if os.path.islink(symlink_path):
-                    os.unlink(symlink_path)
-                os.symlink(source, symlink_path)
-            else:
-                raise PipelineError('Could not find the file ' + source + ', unable to create link')
+        create_output_links(
+            self.job_dir,
+            output_cfg,
+            dir_with_linked_files
+        )
```

Comment:
Also missing `toolset.write_to_yaml` for dependency version reporting. I'm feeling like version reporting and md5s should be a standard part of any data output, so there should be a generic DataOutput class in `pipelines.common` that puts all this alongside link creation and the call to output_data_and_archive.